### PR TITLE
Release 12/12/2017

### PR DESCRIPTION
**Pull request by anthonygarvan**

### CODE REVIEW

File: api/model/form/location.go [hunk 1]
```diff
@@ -310,6 +310,7 @@ var (
 	}
 
 	countries = []string{
+		"POSTOFFICE",
```
Comment (anthonygarvan) [inline api/model/form/location.go]: this seems strange. I'm sure it's not a mistake but just curious, why is this here?

File: src/components/Form/Email/Email.jsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ export default class Email extends ValidationElement {
 
 Email.defaultProps = {
   // pattern: `^([A-z0-9_\.-]+)@([A-z0-9\.-]+)\.+([A-z\.]{2,6})$`,
```
Comment (anthonygarvan) [inline src/components/Form/Email/Email.jsx]: please remove commented code

File: src/components/Form/Email/Email.jsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ export default class Email extends ValidationElement {
...
-  pattern: /^([A-z0-9_\.-]+)@([A-z0-9\.-]+)\.+([A-z\.]{2,6})$/,
+  pattern: /^([A-z0-9_\.-]+)@([A-z0-9\.-]+)\.+([A-z\.]{2,63})$/,
```
Comment (anthonygarvan) [inline src/components/Form/Email/Email.jsx]: are there tests for this regex?

Comment (anthonygarvan) [reply src/components/Form/Email/Email.jsx]: nevermind found them :)

File: src/components/Section/History/dateranges.js [hunk 4]
```diff
@@ -181,92 +173,80 @@ export const gaps = (ranges = [], start = ten, buffer = 30) => {
   const fullStop = today
   const length = ranges.length - 1
 
-  // Set the day to the last of the month
-  start = endOfMonth(start)
+  // // Set the day to the last of the month
+  // start = endOfMonth(start)
```
Comment (anthonygarvan) [inline src/components/Section/History/dateranges.js]: remove commented out code

File: src/components/Section/History/dateranges.js [hunk 5]
```diff
@@ -181,92 +173,80 @@ export const gaps = (ranges = [], start = ten, buffer = 30) => {
...
 
   ranges.sort(rangeSorter).forEach((range, i) => {
     if (!range.from || !range.to) {
       return
     }
 
     // Finds the gaps from the past to the present
-    const stop = new Date(range.from.date)
+    // const stop = new Date(range.from.date)
```
Comment (anthonygarvan) [inline src/components/Section/History/dateranges.js]: remove commented out code

File: src/components/Section/History/dateranges.js [hunk 6]
```diff
@@ -181,92 +173,80 @@ export const gaps = (ranges = [], start = ten, buffer = 30) => {
...
-          date: range.from.date
-        }
-      })
+      holes.push({ from: start, to: range.from })
     }
 
     // Set the next start position
-    start = endOfMonth(range.to.date)
+    // start = endOfMonth(range.to.date)
+    // start = endOfMonth(range.to)
```
Comment (anthonygarvan) [inline src/components/Section/History/dateranges.js]: remove commented out code

File: src/components/Section/History/dateranges.js [hunk 7]
```diff
@@ -181,92 +173,80 @@ export const gaps = (ranges = [], start = ten, buffer = 30) => {
...
+//         } else {
+//           // We now have to shrink the gap
+//           gaps[i].from = range.to
+//         }
+//       }
+//     }
+//   }
+
+//   return gaps.filter(gap => daysBetween(gap.from, gap.to) > buffer)
+// }
```
Comment (anthonygarvan) [inline src/components/Section/History/dateranges.js]: remove commented out code

File: api/model/form/location.go (hunk 1 above)
Comment (bmallred) [reply api/model/form/location.go]: That is the "country" value when the address is a APO/FPO/DPO. 

We can probably look at removing it here and make sure it is not included in the validation.
